packages/amplify-provider-awscloudformation/src/aws-utils/aws-cfn.js

```diff
@@ -384,13 +384,9 @@ class CloudFormation {
         });
     } catch (error) {
       this.progressBar?.stop();
-
-      if (error instanceof AmplifyException) {
-        throw error;
-      }
-
       throw new AmplifyFault('ResourceNotReadyFault', {
-        message: error.message
+        message: error.message,
+        code: error.code,
```

Comment:
Won't this translate any specific `AmplifyException` thrown from the try block above to a `ResourceNotReadyFault`? Is this the expectation?

Reply:
no, in our exception handler we print/report the deepest exception in the chain, the above `IF` check was from the time when we did not have that logic in our handler